lib/msal-common/src/auth/authority/AuthorityFactory.ts

```diff
@@ -4,14 +4,25 @@
  */
 import { Authority } from "./Authority";
 import { AadAuthority } from "./AadAuthority";
+import { B2cAuthority, B2CTrustedHostList } from "./B2cAuthority";
 import { AuthorityType } from "./AuthorityType";
 import { ClientConfigurationError } from "../../error/ClientConfigurationError";
 import { ClientAuthError } from "../../error/ClientAuthError";
 import { INetworkModule } from "../../network/INetworkModule";
 import { StringUtils } from "../../utils/StringUtils";
 import { UrlString } from "../../url/UrlString";
 
-export class AuthorityFactory {
+export class AuthorityFactory {  
+    /**
+     * Use when Authority is B2C to provide list of trusted/allowed domains.
+     */
+    public static setKnownAuthorities(knownAuthorities: Array<string>): void {
+        if (!B2CTrustedHostList.length){
+            knownAuthorities.forEach(function(authority){
+                B2CTrustedHostList.push(authority);
```

Comment:
I would personally prefer some sort of method on the b2cauthority class for adding an authority. This is leaking implementation details.

Reply:
Moved to B2CAuthority class